Dockerfile

```diff
@@ -4,82 +4,69 @@ FROM debian:latest
-    parallel bzip2 acl gnupg
-
 # Install local mail transport agent
-RUN apt-get -y update && apt-get install -y exim4-daemon-light
 COPY update-exim4.conf.conf /etc/exim4/update-exim4.conf.conf
 RUN update-exim4.conf && service exim4 restart
 
 # Set up users and permissions
 RUN adduser --gecos '' --uid 1000 --gid 50 --disabled-password php \
     && adduser php staff
 
-# Install PHP 7 and some common extensions
-RUN wget -O - https://www.dotdeb.org/dotdeb.gpg | apt-key add -
-RUN echo "deb https://packages.dotdeb.org $(lsb_release -sc) all" | tee /etc/apt/sources.list.d/php.list
-RUN echo "deb-src https://packages.dotdeb.org $(lsb_release -sc) all" | tee -a /etc/apt/sources.list.d/php.list
-RUN apt-get -y update \
+# Install PHP 7 and extensions and update settings
+RUN wget -O - https://www.dotdeb.org/dotdeb.gpg | apt-key add - \
+    && echo "deb https://packages.dotdeb.org $(lsb_release -sc) all" | tee /etc/apt/sources.list.d/php.list \
+    && echo "deb-src https://packages.dotdeb.org $(lsb_release -sc) all" | tee -a /etc/apt/sources.list.d/php.list \
+    && apt-get -y update \
     && apt-get install -y php7.0-fpm php7.0-cli php7.0-mbstring php7.0-mcrypt php7.0-intl \
-    php7.0-mysql php7.0-redis php7.0-memcached php7.0-gd php7.0-curl php7.0-xsl
-# Update some PHP settings
-RUN sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/cli/php.ini
-RUN sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/fpm/php.ini
-# Composer updates can require a lot of memory
-RUN sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 4G/" /etc/php/7.0/cli/php.ini
-RUN sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 256M/" /etc/php/7.0/fpm/php.ini
-RUN phpenmod -v 7.0 -s ALL mbstring mcrypt intl pdo_mysql redis memcached gd curl xml xsl
+    php7.0-mysql php7.0-redis php7.0-memcached php7.0-gd php7.0-curl php7.0-xsl php-imagick \
+    && sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/cli/php.ini \
+    && sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/fpm/php.ini \
+    && sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 4G/" /etc/php/7.0/cli/php.ini \
+    && sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 256M/" /etc/php/7.0/fpm/php.ini \
+    && phpenmod -v 7.0 -s ALL mbstring mcrypt intl pdo_mysql redis memcached gd curl xml xsl imagick
 
 # Copy in php-fpm config file
 COPY php-fpm.conf /etc/php/7.0/fpm/php-fpm.conf
 
-# Install composer
+# Install tooling
 RUN wget -O - https://getcomposer.org/installer | php \
     && mv composer.phar /usr/local/bin/composer \
-    && composer self-update
-
-# Install nodejs and npm
-RUN curl -sL https://deb.nodesource.com/setup_6.x | bash - \
-    && apt-get install -y nodejs
-
-# Install nginx and supervisord from dotdeb
-RUN apt-get -y update && apt-get -y install supervisor nginx
-
-# One last update for everything
-RUN apt-get -y update && apt-get -y dist-upgrade
+    && composer self-update && \
```

Comment:
Not really needed, because you are installing a fresh copy.
Also check this: https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md
Something with a checksum

Reply:
I'm taking a closer look at the composer docs.

Reply:
Added checking the signature of the installer.